episodes/11-parallel-intro.md

```diff
@@ -0,0 +1,101 @@
+
+::::::::::::::::::::::::::::::::::::: objectives
+- "First objective."
+::::::::::::::::::::::::::::::::::::::::::::::::
+
+The basic concept of parallel computing is simple to understand: we divide our job into tasks that can be
+executed at the same time, so that we finish the job in a fraction of the time that it would have taken if the
+tasks were executed one by one. Implementing parallel computations, however, is not always easy, nor
+possible...
+
+Consider the following analogy:
+
+Suppose that we want to paint the four walls in a room. We'll call this the *problem*. We can divide our
+problem into 4 different tasks: paint each of the walls. In principle, our 4 tasks are independent from each
+other in the sense that we don't need to finish one to start one another. We say that we have 4 **_concurrent
+tasks_**; the tasks can be executed within the same time frame.  However, this does not mean that the tasks
+can be executed simultaneously or in parallel. It all depends on the amount of resources that we have for the
+tasks.  If there is only one painter, this guy could work for a while in one wall, then start painting another
+one, then work for a little bit on the third one, and so on. **_The tasks are being executed concurrently but
+not in parallel_**. If we have two painters for the job, then more parallelism can be introduced. Four
+painters could executed the tasks **_truly in parallel_**.
+
+::::::::::::::::::::::::::::::::::::: callout
+
+Think of the CPU cores as the painters or workers that will execute your concurrent tasks
+
+::::::::::::::::::::::::::::::::::::::::::::::::
+
+Now imagine that all workers have to obtain their paint from a central dispenser located at the middle of the
+room. If each worker is using a different colour, then they can work **_asynchronously_**, however, if they
+use the same colour, and two of them run out of paint at the same time, then they have to **_synchronise_** to
+use the dispenser: One must wait while the other is being serviced.
+
+::::::::::::::::::::::::::::::::::::: callout
+
+Think of the shared memory in your computer as the central dispenser for all your workers
+
+::::::::::::::::::::::::::::::::::::::::::::::::
+
+Finally, imagine that we have 4 paint dispensers, one for each worker. In this scenario, each worker can
+complete their task totally on their own. They don't even have to be in the same room, they could be painting
+walls of different rooms in the house, in different houses in the city, and different cities in the
+country. We need, however, a communication system in place. Suppose that worker A, for some reason, needs a
+colour that is only available in the dispenser of worker B, they must then synchronise: worker A must request
+the paint of worker B and worker B must respond by sending the required colour.
+
+::::::::::::::::::::::::::::::::::::: callout
+
+Think of the memory distributed on each node of a cluster as the different dispensers for your workers
```

Comment:
This requires a lot more explanation. Has **_distributed memory_** been introduced? It's probably OK, but the analogy here should be clarified.

Reply:
There is no need to introduce *distributed memory* at this point yet. I reformulated the analogy with multiple dispensers without using the word "distributed". Resolved through a commit.